internal/client/gandi.go

```diff
@@ -27,15 +28,20 @@ type Gandi struct {
 }
 
 // New instantiates a new Gandi client
-func New(apikey string, apiurl string, sharingID string, debug bool, dryRun bool, timeout time.Duration) *Gandi {
+func New(apikey string, token string, apiurl string, sharingID string, debug bool, dryRun bool, timeout time.Duration) *Gandi {
 	if apiurl == "" {
 		apiurl = config.APIURL
 	}
 	endpoint := apiurl + "/v5/"
 	if timeout == 0 {
 		timeout = config.Timeout
 	}
-	return &Gandi{apikey: apikey, endpoint: endpoint, sharingID: sharingID, debug: debug, dryRun: dryRun, timeout: timeout}
+	if apikey != "" {
+		log.Println("Warning: Authenticating using an API key is deprecated - please use a Personal Access Token instead.")
```

Comment:
Question: Do we _actually_ want to be printing warnings here? Might get quite annoying for downstream users. Perhaps either something with a more advanced logger (:shrug: ), or we bite the bullet and port everything to returning error types too.

Reply:
Yes, i agree with you that it could be annoying for downstream users! (sorry for having proposed this :/)

I think it would be enough to mention in the documentation the API key is deprecated. Downstream users could then decide if they want to expose this mecanism to their users.